tools/importer-rest-api-specs/components/parser/models_discriminators_test.go

```diff
@@ -1359,3 +1359,66 @@ func TestParseDiscriminatorsOrphanedChildWithNestedModel(t *testing.T) {
 	}
 	validateParsedSwaggerResultMatches(t, expected, actual)
 }
+
+func TestParseDiscriminatorsOrphanedChildWithoutDiscriminatorValue(t *testing.T) {
+	actual, err := ParseSwaggerFileForTesting(t, "/nestedtestdata/model_discriminators_orphaned_child_without_discriminator_value.json", pointer.To("datafactory"))
```

Comment:
Since this functionality is scoped to DataFactory, perhaps it's worth a second test with the same swagger file for `Compute` to sanity-check that we don't unintentionally enable this elsewhere - WDYT?